java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/launch/NbLaunchDelegate.java

```diff
@@ -291,14 +291,14 @@ public void propertyChange(PropertyChangeEvent evt) {
                             }
                         }
                     });
-                } else {
-                    launchFuture.complete(null);
                 }
-                
                 Lookups.executeWith(launchCtx, () -> {
                     providerAndCommand.first().invokeAction(providerAndCommand.second(), lookup);
```

Comment:
But the `invokeAction` returns immediately without waiting for the action to finish!

If you want to wait for the action to fully finish, you have to use `ActionProgress.finish(...)`.

Reply:
There is no need for waiting for action to finish but rather to ensure the action has already started.

Reply:
Alas, the [VSNetBeans integration tests](https://app.travis-ci.com/github/apache/netbeans/jobs/524067665) continue to fail.